File: BE/DACN/src/main/java/com/haui/service/WishlistService.java
package com.haui.service;

import com.haui.dto.response.wishlist.WishlistDto;
import com.haui.dto.response.wishlist.product.WishlistProductDto;
import com.haui.dto.response.wishlist.product.WishlistItemDto;

import java.util.List;

public interface WishlistService {
    WishlistDto getByUserId(Integer userId);

    List<WishlistItemDto> getProducts(Integer userId);

    void addProduct(Integer userId, Integer productId);

    void removeProduct(Integer userId, Integer productId);

    WishlistDto create(Integer userId);
}

File: FE/laptopshop/services/wishlist/WishlistService.ts
import { clientApi } from "@/lib/axios/client";
import { WishlistDto, WishlistItemDto } from "@/types/wishlist/wishlist";

export const wishlistService = {
  // GET /wishlists
  getMyWishlist: async (): Promise<WishlistDto> => {
    const res = await clientApi.get(`/wishlists`);
    return res.data.data;
  },

  // POST /wishlists
  create: async (): Promise<WishlistDto> => {
    const res = await clientApi.post(`/wishlists`);
    return res.data.data;
  },

  // GET /wishlists/products
  getProductWishlist: async (): Promise<WishlistItemDto[]> => {
    const res = await clientApi.get(`/wishlists/products`);
    return res.data.data;
  },

  // POST /wishlists/products?productId=...
  addProduct: async (productId: number): Promise<void> => {
    await clientApi.post(`/wishlists/products`, null, {
      params: { productId },
    });
  },

  // DELETE /wishlists/products/{productId}
  removeProduct: async (productId: number): Promise<void> => {
    await clientApi.delete(`/wishlists/products/${productId}`);
  },
};
